### PR TITLE
Remove auto-downing, #27788

### PR DESCRIPTION
**Pull request by patriknw**
* moved to cluster tests, in new package akka.cluster.testkit
* changed config in tests
* migration guide
* documentation clarificiations for Downing and Leaving
* update warnings in Singleton and Sharding

Refs #27788

@TimMoore could you review the documentation changes here

### CODE REVIEW

File: akka-cluster/src/main/scala/akka/cluster/ClusterSettings.scala [hunk 1]
```diff
@@ -116,21 +116,6 @@ final class ClusterSettings(val config: Config, val systemName: String) {
...
-      case _     => cc.getMillisDuration(key).requiring(_ >= Duration.Zero, key + " >= 0s, or off")
-    }
-  }
-
-  /**
-   * @deprecated Specific to [[akka.cluster.AutoDown]] should not be used anywhere else, instead
-   *   ``Cluster.downingProvider.downRemovalMargin`` should be used as it allows the downing provider to decide removal
-   *   margins
-   */
-  @deprecated("Use Cluster.downingProvider.downRemovalMargin", since = "2.4.5")
```
Comment (johanandren) [inline akka-cluster/src/main/scala/akka/cluster/ClusterSettings.scala]: Isn't the deprecation still valid?

File: akka-cluster/src/test/scala/akka/cluster/testkit/AutoDown.scala [hunk 2]
```diff
@@ -2,49 +2,94 @@
...
+ * Downing provider used for testing.
+ *
+ * Auto-downing is a naïve approach to remove unreachable nodes from the cluster membership.
+ * In a production environment it will eventually breakdown the cluster.
+ * When a network partition occurs, both sides of the partition will see the other side as unreachable
+ * and remove it from the cluster. This results in the formation of two separate, disconnected, clusters
+ * (known as *Split Brain*).
+ *
+ * This behaviour is not limited to network partitions. It can also occur if a node in the cluster is
+ * overloaded, or experiences a long GC pause.
```
Comment (johanandren) [inline akka-cluster/src/test/scala/akka/cluster/testkit/AutoDown.scala]: surprise brittish spelling of behavior

File: akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md [hunk 3]
```diff
@@ -11,6 +11,40 @@ is [no longer available as a static method](https://github.com/scala/bug/issues/
...
+## Auto-downing removed
+
+Auto-downing of unreachable Cluster members have been removed after warnings and recommendations against using it
+for many years. It was by default disabled, but could be enabled with configuration
+`akka.cluster.auto-down-unreachable-after`.
+
+For alternatives see the @ref:[documentation about Downing](../typed/cluster.md#downing).
+
+Auto-downing was a naïve approach to remove unreachable nodes from the cluster membership.
+In a production environment it will eventually breakdown the cluster.
```
Comment (johanandren) [inline akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md]: break down

**Review by johanandren — APPROVED** [review]
LGTM with some spelling/language nits

File: akka-docs/src/main/paradox/typed/cluster.md [hunk 4]
```diff
@@ -255,95 +255,68 @@ after the restart, when it come up as new incarnation of existing member in the
...
 
-@@@
+Note that @ref:[Cluster Singleton](cluster-singleton.md) or @ref:[Cluster Sharding entities](cluster-sharding.md) that
+are running on a crashed (unreachable) node will not be started on another node until the previous node has
+been removed from the Cluster. Removal of crashed (unreachable) nodes is performed after a downing decision.
 
-### Leaving
+A production solution for downing is provided by
+[Split Brain Resolver](https://doc.akka.io/docs/akka-enhancements/current/split-brain-resolver.html),
+which is part of the [Lightbend Reactive Platform](http://www.lightbend.com/platform).
```
Comment (TimMoore) [inline akka-docs/src/main/paradox/typed/cluster.md]:
```suggestion
which is part of the [Lightbend Platform](https://www.lightbend.com/lightbend-platform).
```

File: akka-docs/src/main/paradox/typed/cluster.md [hunk 5]
```diff
@@ -255,95 +255,68 @@ after the restart, when it come up as new incarnation of existing member in the
...
+If you don’t use RP, you should anyway carefully read the [documentation](https://doc.akka.io/docs/akka-enhancements/current/split-brain-resolver.html)
```
Comment (TimMoore) [inline akka-docs/src/main/paradox/typed/cluster.md]:
```suggestion
If you don’t have a Lightbend Platform Subscription, you should still carefully read the [documentation](https://doc.akka.io/docs/akka-enhancements/current/split-brain-resolver.html)
```

**Review by TimMoore — COMMENTED** [review]
Just a couple small comments. Looks great.

File: akka-cluster/src/main/scala/akka/cluster/ClusterSettings.scala (hunk 1 above)
Comment (patriknw) [reply akka-cluster/src/main/scala/akka/cluster/ClusterSettings.scala]: A downing provider can define a `downRemovalMargin`, but if no downing provider is used this is still useful to have as a possible config property. It is used from the `NoDowning` provider and also from ClusterSingleton. I think we should just revert the decision to deprecate.

File: akka-docs/src/main/paradox/typed/cluster.md (hunk 4 above)
Comment (patriknw) [reply akka-docs/src/main/paradox/typed/cluster.md]: also changed at one more place

**Comment by patriknw** [comment]
incorporated review feedback

**Review by helena — APPROVED** [review]
The Akka code base just lost 10 pounds.

**Comment by patriknw** [comment]
I'll run this in jenkins multi-jvm job before we merge

**Review by chbatey — APPROVED** [review]
:+1:

**Comment by patriknw** [comment]
all multi node tests passed

**Pull request merged**